cirq-core/cirq/circuits/moment.py

```diff
@@ -91,6 +91,7 @@ def __init__(self, *contents: 'cirq.OP_TREE') -> None:
             ValueError: A qubit appears more than once.
         """
         self._operations = tuple(op_tree.flatten_to_ops(contents))
+        self._sorted_operations = tuple(sorted(self._operations, key=lambda op: op.qubits))
```

Comment:
I'm a bit worried about hit we take here for time and memory.  Already our circuit code isn't particularly fast and this doubles the memory and time for constructing moments.  

Does it make sense to maybe have this be a cached value (like measurement key obs and control keys) that is only populated when used?

Reply:
done, although I think it would have little value since sorted operations is used in common functions like `to_text_diagram` and `_eq_`.